code/modules/mob/mob_helpers.dm

```diff
@@ -388,24 +388,36 @@ var/list/intents = list(INTENT_HELP,INTENT_DISARM,INTENT_GRAB,INTENT_HARM)
 	set name = "Sleep"
 	set category = "IC"
 
-	if(sleeping)
+	if(IsSleeping())
 		to_chat(src, "<span class='notice'>You are already sleeping.</span>")
 		return
 	else
 		if(alert(src, "You sure you want to sleep for a while?", "Sleep", "Yes", "No") == "Yes")
-			SetSleeping(20) //Short nap
+			SetSleeping(400) //Short nap
 
 /mob/living/verb/lay_down()
 	set name = "Rest"
 	set category = "IC"
 
 	if(!resting)
-		client.move_delay = world.time + 20
-		to_chat(src, "<span class='notice'>You are now resting.</span>")
-		StartResting()
+		set_resting(TRUE, FALSE)
 	else if(resting)
-		to_chat(src, "<span class='notice'>You are now getting up.</span>")
-		StopResting()
+		if(do_after(src, 10, 0, src, 1, null, MOBILITY_MOVE))

```

Comment:
resting now has an do_after

Reply:
RIP matrixing under lasers and bullets once again 🤣

Reply:
We'll have the issue of people crawling around, as bullets won't hit them. Til the maints or staff tell me how they wanna go with that at least. It wouldn't be hard to make bullets hit people regardless of resting or not but I didn't include any of those changes in this PR so you can crawl around to dodge bullets at the cost of being slower.

Reply:
Does this PR add the ability to crawl voluntarily while rested, or only while stunned?
It makes sense that crawling to avoid projectiles lazily aimed for center of mass. Perhaps down the road it could have some rng to see whether the shot 'dipped' to hit a crawler as it passed the turf.

Reply:
It adds in the ability to crawl voluntarily while rested. You can even use items while rested there's no restrictions currently. Though I think a nice compromise to the system between "no you can't duck under bullets" and "yes you can lie down and machine gun people" would just be to restrict gun/ranged weapon usage on the floor, and then just having it as a melee thing. of course you could still throw stuff but it'd be cool to see someone with a stun baton slide under some bullets and then whack someone

Reply:
Lmfao! Great. Thanks for the quick response, it would be pretty badass-- but that's assuming the guy doesn't actually click you dead-on to score actual hits 😄 not something I myself can pull off but I know there's sharpshooters among us

Reply:
Would make the moment even better considering that would even be a possibility. No problem with the responses it's fun talking to someone about code and stuff. Feel free to comment on any of my other PRs or anything and i'll be happy to reply when I can get to it